File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel/Persistence/CdmFolder/Types/ManifestDeclaration.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="ManifestDeclaration.cs" company="Microsoft">
//      All rights reserved.
// </copyright>
// <summary>
//   The folder declaration for CDM manifest format.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace Microsoft.CommonDataModel.ObjectModel.Persistence.CdmFolder.Types
{
    /// <summary>
    /// The manifest declaration for CDM manifest format.
    /// </summary>
    public class ManifestDeclaration : FileStatus
    {
        /// <summary>
        /// Gets or sets the explanation.
        /// </summary>
        public string Explanation { get; set; }

        /// <summary>
        /// Gets or sets the manifest name.
        /// </summary>
        public string ManifestName { get; set; }

        /// <summary>
        /// Gets or sets the corpus path to the definition of the sub manifest.
        /// </summary>
        public string Definition { get; set; }

        /// <inheritdoc />
        public string LastFileStatusCheckTime { get; set; }

        /// <inheritdoc />
        public string LastFileModifiedTime { get; set; }

        /// <inheritdoc />
        public string LastChildFileModifiedTime { get; set; }
    }
}

File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/persistence/cdmfolder/types/ManifestDeclaration.java
package com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.types;

public class ManifestDeclaration extends FileStatusExtended {
  private String explanation;
  private String manifestName;
  private String definition;

  public String getExplanation() {
    return explanation;
  }

  public void setExplanation(final String explanation) {
    this.explanation = explanation;
  }

  public String getManifestName() {
    return manifestName;
  }

  public void setManifestName(final String manifestName) {
    this.manifestName = manifestName;
  }

  //    TODO-BQ: 2019-09-26 Temporary support for FolioName
  public void setFolioName(final String folioName) {
    this.manifestName = folioName;
  }

  /**
   * Gets the corpus path to the definition of the sub manifest.
   *
   * @return
   */
  public String getDefinition() {
    return definition;
  }

  /**
   * Sets the corpus path to the definition of the sub manifest.
   *
   * @param definition
   */
  public void setDefinition(final String definition) {
    this.definition = definition;
  }
}